### ISSUE TITLE
Video URL without extension not reproducing on iOS

### ISSUE THREAD

**Issue by alemosa** [vidstack/player#1166]
### Current Behavior:
Given videoUrl is pointing to a video file without extension, like _https://pull-zone.b-cdn.net/videoId/original_
When setting up a video source with src={{src: videoUrl, type: "video/mp4"}} I expect it to be available both on Desktop and Mobile, but in mobile phone (tested on iOS) it's currently not working.

Here you can find a demo: https://video-tests-six.vercel.app/

### Expected Behavior:

In mobile, it should be able to reproduce the source without issues, like an HTML Video Player does.

### Steps To Reproduce:

Visit  https://video-tests-six.vercel.app/ from a mobile/iOS device, click on play, vidstack player not working. If you test with HTML, video works.

**Reproduction Link:** [How to create a repro?][repro]

[repro]:  https://video-tests-six.vercel.app/

### Environment:

- Framework: React
- Meta Framework: Next.js
- Node: 16.0.0
- Device: iPhone@13
- OS: iOS@14
- Browser: Chrome@22

**Comment by mihar-22** [comment]
I tested this video with the native HTML `<video>` element in iOS Safari and it doesn't work. The browser is returning a code `4` which means the source is not supported. This might be from an unsupported codec, invalid MP4 container, HTTP range requests, cross-origin issues, or the file is corrupted. I checked and everything seems fine which is strange, without digging further my best guess is it might be an invalid MP4 container (according to iOS Safari).
